packages/storage/src/service.ts

```diff
@@ -133,7 +144,8 @@ function extractBucket(config?: FirebaseOptions): Location | null {
  * @param opt_url - gs:// url to a custom Storage Bucket
  */
 export class StorageService implements _FirebaseService {
-  readonly _bucket: Location | null = null;
+  _bucket: Location | null = null;
+  private _emulatorOrigin?: string;
```

Comment:
Could this just be `host` and be initialized to DEFAULT_HOST? I think this would get rid of most "is host undefined" checks.

Reply:
Makes sense, done.